partiql-logical-planner/src/lower.rs

```diff
@@ -361,6 +361,18 @@ impl<'a> AstToLogical<'a> {
                                                     lookups.push(expr);
                                                 }
                                                 continue;
+                                            } else if let Some(_type_entry) = self
```

Comment:
Could we add a test case showing the new name resolution rule chooses the catalog entry?

Reply:
I have been depending on the conformance tests for the correction of this and since there had been no regression and this is not baked in the evaluator yet, decided to exclude the test. I will add a test that inspects the `LogicalPlan` as @alancai98 suggested in our offline chat.